backend/app/services/cache.py

```
@@ -25,16 +27,16 @@ def generate_search_key(
+        limit: int,
     ) -> str:
         """Generate cache key for search results"""
         key_data = {
             "tenant_id": tenant_id,
             "project_id": project_id,
             "query_hash": query_hash,
             "filters": sorted(filters.items()),
-            "limit": limit
+            "limit": limit,
         }
         return f"search:{hashlib.md5(json.dumps(key_data, sort_keys=True).encode()).hexdigest()}"
```

Ruff (0.13.1)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)